go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -235,6 +239,11 @@ func localnetAddService(svc *kapi.Service) error {
 }
 
 func localnetDeleteService(svc *kapi.Service) error {
+	for _, externalIP := range svc.Spec.ExternalIPs {
```

Comment:
While this will be correct most of the time, it won't handle the case where an ExternalIP is deleted while the process is restarting. It would be good to ad an additional synchronization mechanism that caught this (though it is unlikely).

Reply:
Indeed, I agree. However I think that's a generic flaw with ovn-kubernetes and should maybe be fixed with a dedicated PR.

Reply:
in this case, ovn-kubernetes has some tooling to deal with this - when you add the handler, you can also add a "sync" function that gets the full list of services. There, you can delete any rules that shouldn't exist.